docs/csharp/language-reference/keywords/extern.md

```diff
@@ -10,80 +10,83 @@ helpviewer_keywords:
-public class MainClass   
-{  
-   [DllImport("Cmdll.dll")]  
-   public static extern int SampleMethod(int x);  
-  
-   static void Main()   
-   {  
-      Console.WriteLine("SampleMethod() returns {0}.", SampleMethod(5));  
-   }  
-}  
-```  
-  
-## Example  
- 3. Open a Visual Studio x64 (or x32) Native Tools Command Prompt window from the Visual Studio installation directory and compile the `cm.cs` file by typing:  
-  
-> **csc cm.cs** (for the x64 command prompt)   
->  —or—  
-> **csc /platform:x86 cm.cs** (for the x32 command prompt)  
-  
- This will create the executable file `cm.exe`.  
-  
- 4. Run `cm.exe`. The `SampleMethod` method passes the value 5 to the DLL file, which returns the value multiplied by 10.  The program produces the following output:  
-  
-```  
-SampleMethod() returns 50.  
-```  
-  
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
-  
-## See Also  
- <xref:System.Runtime.InteropServices.DllImportAttribute?displayProperty=nameWithType>  
- [C# Reference](../../../csharp/language-reference/index.md)  
- [C# Programming Guide](../../../csharp/programming-guide/index.md)  
- [C# Keywords](../../../csharp/language-reference/keywords/index.md)  
- [Modifiers](../../../csharp/language-reference/keywords/modifiers.md)
+
+The `extern` modifier is used to declare a method that is implemented externally. A common use of the `extern` modifier is with the `DllImport` attribute when you are using Interop services to call into unmanaged code. In this case, the method must also be declared as `static`, as shown in the following example:
+
+```csharp
+[DllImport("avifil32.dll")]
+private static extern void AVIFileInit();
+```
+
+The `extern` keyword can also define an external assembly alias, which makes it possible to reference different versions of the same component from within a single assembly. For more information, see [extern alias](extern-alias.md).
+
+It is an error to use the [abstract](abstract.md) and `extern` modifiers together to modify the same member. Using the `extern` modifier means that the method is implemented outside the C# code, whereas using the `abstract` modifier means that the method implementation is not provided in the class.
+
+The extern keyword has more limited uses in C# than in C++. To compare the C# keyword with the C++ keyword, see Using extern to Specify Linkage in the C++ Language Reference.
+
+## Example 1
+
+In this example, the program receives a string from the user and displays it inside a message box. The program uses the `MessageBox` method imported from the User32.dll library.
+
+[!code-csharp[csrefKeywordsModifiers#8](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsModifiers/CS/csrefKeywordsModifiers.cs#8)]
+
+## Example 2
+
+This example illustrates a C# program that calls into a C library (a native DLL).
+
+1. Create the following C file and name it `cmdll.c`:
+
+```c
+// cmdll.c
+// Compile with: /LD
```

Comment:
not related to your PR: We've been trying to use the '-' for options instead of '/' for platform independence.